### PR TITLE
add documentation for $pattern annotation 

### LINKED ISSUE #626: Add doc for `$pattern`
We are adding annotation `$pattern` in https://github.com/SAP/abap-file-formats-tools/pull/312 and need an example/documentation here.

### PR DESCRIPTION
**Pull request by jonassobotta**
This adds documentation for the new annotation `$pattern` in https://github.com/SAP/abap-file-formats-tools/pull/312 

Fixes #626

### CODE REVIEW

File: docs/json.md [hunk 1]
```diff
@@ -197,6 +197,13 @@ Remark: If an enum is used, it should be checked if one of the following points
 
 In case additional values for the enum should be added compatibly later, a default value must always be specified (see [Format Versions and Compatibility](#format-versions-and-compatibility)). If systems don't support the new enumeration value (e.g., in lower releases), the value will be changed to the default value by the file format implementations.
 
+### Regular Expressions
+
+In order to express constraints to the values of a field, it is possible to pass [regualr expressions](https://json-schema.org/understanding-json-schema/reference/regular_expressions) enclosed in single quotes:
```
Comment (albertmink) [inline docs/json.md:202]:
```suggestion
In order to express constraints to the values of a field, it is possible to pass [regular expressions](https://json-schema.org/understanding-json-schema/reference/regular_expressions) enclosed in single quotes:
```

**Review by albertmink — CHANGES_REQUESTED** [review]
Found a typo. And why don't you put the new section in between `Length Specification` and `Mutliple Of`? Right now, it has been added just in the middle of section `Enum Values`.

File: docs/json.md (hunk 1 above)
Comment (jonassobotta) [reply docs/json.md]: Thanks for the hint. I moved the section in between `Length Specification` and `Mutliple Of`. I simply oversaw that it was in the middle of the `Enum Values`.

**Review by albertmink — APPROVED** [review]
Thanks, LGTM

**Pull request merged**